File: CS/DataNavigatorInBar/CustomRepositoryItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Repository;
using System.Windows.Forms;
using DevExpress.XtraEditors.ViewInfo;

namespace DataNavigatorInBar
{
    [UserRepositoryItem("Register")]
    class NavigatorEditRepositoryItem:RepositoryItem
    {
        MyDataNavigator _drawNavigator;
        public MyDataNavigator DrawNavigator
        {
            get { return _drawNavigator;}
            set { _drawNavigator = value;}
        }
        MyDataNavigator _editorNavigator;
        public MyDataNavigator EditorNavigator
        {
            get { return _editorNavigator; }
            set { _editorNavigator = value; }
        }

        static NavigatorEditRepositoryItem()
        {
            Register();
        }
        public NavigatorEditRepositoryItem() :base()
        {
            _drawNavigator = new MyDataNavigator();
            _drawNavigator.Dock = DockStyle.Fill;
            _editorNavigator = new MyDataNavigator();
            _editorNavigator.Dock = DockStyle.Fill;
            _drawNavigator.BindingContext = new BindingContext();

        }
        internal const string EditorName = "DataNavigatorEdit";

        public static void Register()
        {
            EditorRegistrationInfo.Default.Editors.Add(new EditorClassInfo(EditorName, typeof(NavigatorEdit),
                typeof(NavigatorEditRepositoryItem), typeof(BaseEditViewInfo),
                new NavigatorEditPainter(), true));
        }
        public override string EditorTypeName
        {
            get { return EditorName; }
        }
    }

}

File: VB/DataNavigatorInBar/CustomRepositoryItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel

Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Repository
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.ViewInfo

Namespace DataNavigatorInBar
	<UserRepositoryItem("Register")> _
	Friend Class NavigatorEditRepositoryItem
		Inherits RepositoryItem
		Private _drawNavigator As MyDataNavigator
		Public Property DrawNavigator() As MyDataNavigator
			Get
				Return _drawNavigator
			End Get
			Set(ByVal value As MyDataNavigator)
				_drawNavigator = value
			End Set
		End Property
		Private _editorNavigator As MyDataNavigator
		Public Property EditorNavigator() As MyDataNavigator
			Get
				Return _editorNavigator
			End Get
			Set(ByVal value As MyDataNavigator)
				_editorNavigator = value
			End Set
		End Property

		Shared Sub New()
			Register()
		End Sub
		Public Sub New()
			MyBase.New()
			_drawNavigator = New MyDataNavigator()
			_drawNavigator.Dock = DockStyle.Fill
			_editorNavigator = New MyDataNavigator()
			_editorNavigator.Dock = DockStyle.Fill
			_drawNavigator.BindingContext = New BindingContext()

		End Sub
		Friend Const EditorName As String = "DataNavigatorEdit"

		Public Shared Sub Register()
			EditorRegistrationInfo.Default.Editors.Add(New EditorClassInfo(EditorName, GetType(NavigatorEdit), GetType(NavigatorEditRepositoryItem), GetType(BaseEditViewInfo), New NavigatorEditPainter(), True))
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return EditorName
			End Get
		End Property
	End Class

End Namespace
